### PR TITLE
LazyTensor -> LinearOperator

### PR DESCRIPTION
**Pull request by gpleiss**
The LazyTensor functionality is moving to a separate repository (GitHub.com/cornellius-gp/linear_operator).

This PR deprecates LazyTensor and instead uses the LinearOperator implementation.

(Note that this PR does not change any GPyTorch code; other than removing gpytorch/lazy/*_lazy_tensor.py and all other related files. A separate PR will make changes to the code that rename all *LazyTensor references and utilize the torch_function functionality of LinearOperator).

## New requirements
- Python 3.8 (was 3.7)
- Torch 1.11 (was 1.9/1.10)

### CODE REVIEW

**Comment by Balandat** [comment]
The awesomeness of this is totally awesome.

File: setup.py
Comment (Balandat) [inline setup.py:28]: Is PyTorch 1.11 a requirement to properly implement the LinearOperator functionality? Or can we deal with 1.10 also?

The current requirements would mean skipping the 1.10 release as a min requirement. This is a pretty aggressive version requirement that could make it hard for some people in production settings to use this.

It's a major update of gpytorch and so it's fine, but if we can avoid it then that may make sense.

File: docs/requirements.txt
Comment (Balandat) [inline docs/requirements.txt:7]: Side note: We probably want to update the doc stack at some point and support newer sphinx versions etc. But that's orthogonal to this PR.

**Review by Balandat — COMMENTED** [review]
Looks like thew newest sklearn versions already deprecated py3.7. We could either pin sklearn, or move aggressively on python deprecation and also require 3.8 ourselves. Thoughts?

File: setup.py
Comment (gpleiss) [reply setup.py:28]: The new linear_operator package makes use of some 1.11 naming conventions/helper methods.

Though linear_operator could easily be downgraded, I think the best solution is: 1) cut a release before we merge this PR in, where we bump up the version to 1.10. Then 2) when we merge this PR, we bump to 1.11.

File: docs/requirements.txt
Comment (gpleiss) [reply docs/requirements.txt:7]: Agreed!

**Comment by gpleiss** [comment]
This is no longer a WIP PR, but let's not merge it until we merge #2037 (and put out a 1.7 release).

File: README.md
Comment (Balandat) [inline README.md:23]: Should add linear_operator here (the `main` github branch for now until we have made a proper release).

File: docs/requirements.txt
Comment (Balandat) [inline docs/requirements.txt:9]: why is the jinja pinning necessary here?

File: gpytorch/utils/broadcasting.py
Comment (Balandat) [inline gpytorch/utils/broadcasting.py:19]: Should re remove `_mul_broadcast_shape` from linear_operator now that there is torch.broadcast_shapes?

File: gpytorch/functions/__init__.py
Comment (Balandat) [inline gpytorch/functions/__init__.py:58]:
```suggestion
    raise AttributeError(f"module gpytorch.functions has no attribute {name}.")
```

File: requirements.txt
Comment (Balandat) [inline requirements.txt:2]:
```suggestion
```

File: requirements.txt
Comment (Balandat) [inline requirements.txt:4]: should be able to get rid of this also b/c linear_operator depends on scipy
```suggestion
```

File: setup.py
Comment (Balandat) [inline setup.py:32]:
```suggestion
    "scikit-learn",
```

**Review by Balandat — APPROVED** [review]
This looks great. I didn't read every single character here but overall this should be good to go!

File: README.md
Comment (gpleiss) [reply README.md:22]: We've made a proper release now, so it should be fine.

File: docs/requirements.txt
Comment (gpleiss) [reply docs/requirements.txt]: Hopefully it's not, now that our doc stack has been updated.

File: gpytorch/utils/broadcasting.py
Comment (gpleiss) [reply gpytorch/utils/broadcasting.py:19]: Already done :)

File: gpytorch/__init__.py
Comment (gpleiss) [reply gpytorch/__init__.py:32]: Done!

**Pull request merged**
